packages/shared/src/hooks/createContextAndHook.ts

```diff
@@ -18,16 +21,16 @@ type UseCtxFn<T> = () => T;
  */
 export const createContextAndHook = <CtxVal>(
   displayName: string,
-  options?: Options,
+  options?: Options<Context<CtxVal>>,
```

Comment:
It doesn't appear that `options` is actually used in the codebase. So we could remove this?